ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/keys/KeyUniquenessValidation.scala

```diff
@@ -8,51 +8,35 @@ import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlStateKey
 import com.daml.ledger.participant.state.kvutils.committer.transaction.TransactionCommitter.globalKey
 import com.daml.ledger.participant.state.kvutils.committer.transaction.keys.ContractKeysValidation.{
   Duplicate,
-  KeyValidationState,
-  KeyValidationStatus,
+  KeyValidationError,
 }
 import com.daml.lf.transaction.{Node, NodeId}
 import com.daml.lf.value.Value.ContractId
 
 private[keys] object KeyUniquenessValidation {
+
   def checkNodeKeyUniqueness(
       node: Node.GenNode[NodeId, ContractId],
-      keyValidationState: KeyValidationState,
-  ): KeyValidationStatus =
-    keyValidationState match {
-      case UniquenessKeyValidationState(activeStateKeys) =>
-        node match {
-          case exercise: Node.NodeExercises[NodeId, ContractId]
-              if exercise.key.isDefined && exercise.consuming =>
-            val stateKey = Conversions.globalKeyToStateKey(
-              globalKey(exercise.templateId, exercise.key.get.key)
-            )
-            Right(
-              keyValidationState + KeyValidationState(
-                activeStateKeys = activeStateKeys - stateKey
-              )
-            )
+      activeStateKeys: Set[DamlStateKey],
+  ): Either[KeyValidationError, Set[DamlStateKey]] =
+    node match {
+      case exercise: Node.NodeExercises[NodeId, ContractId]
+          if exercise.key.isDefined && exercise.consuming =>
+        val stateKey =
+          Conversions.globalKeyToStateKey(globalKey(exercise.templateId, exercise.key.get.key))
+        Right(
+          activeStateKeys - stateKey
+        )
```

Comment:
This should also fit on a single line.

Reply:
cff0b3d04370fe63bf24b9bd93cf79edac9c5b51 (still mad at scalafmt)

Reply:
I guess the authors don't want to get into religious wars about newlines